src/main/groovy/com/github/jengelman/gradle/plugins/shadow/transformers/Log4j2PluginsCacheFileTransformer.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License") you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.github.jengelman.gradle.plugins.shadow.transformers
+
+import com.github.jengelman.gradle.plugins.shadow.ShadowStats
+import com.github.jengelman.gradle.plugins.shadow.relocation.RelocateClassContext
+import com.github.jengelman.gradle.plugins.shadow.relocation.Relocator
+import org.apache.commons.io.IOUtils
+import org.apache.commons.io.output.CloseShieldOutputStream
+import org.apache.logging.log4j.core.config.plugins.processor.PluginCache
+import org.apache.logging.log4j.core.config.plugins.processor.PluginEntry
+import org.apache.tools.zip.ZipEntry
+import org.apache.tools.zip.ZipOutputStream
+import org.gradle.api.file.FileTreeElement
+
+import static org.apache.logging.log4j.core.config.plugins.processor.PluginProcessor.PLUGIN_CACHE_FILE;
+
+/**
+ * Modified from the maven equivalent to work with gradle
+ *
+ * @author Paul Nelson Baker
+ * @see <a href="https://www.linkedin.com/in/paul-n-baker/">LinkedIn</a>
+ * @see <a href="https://github.com/paul-nelson-baker/">GitHub</a>
+ * @see <a href="https://github.com/edwgiz/maven-shaded-log4j-transformer">edwgiz/maven-shaded-log4j-transformer</a>
+ * @see <a href="https://github.com/edwgiz/maven-shaded-log4j-transformer/blob/master/src/main/java/com/github/edwgiz/mavenShadePlugin/log4j2CacheTransformer/PluginsCacheFileTransformer.java">PluginsCacheFileTransformer.java</a>
+ */
+class Log4j2PluginsCacheFileTransformer implements Transformer {
+
+    private final List<File> temporaryFiles;
+    private final List<Relocator> relocators;
+
+    public Log4j2PluginsCacheFileTransformer() {
+        temporaryFiles = new ArrayList<>();
+        relocators = new ArrayList<>();
+    }
+
+    @Override
+    boolean canTransformResource(FileTreeElement element) {
+        return PLUGIN_CACHE_FILE == element.name
```

Comment:
This line does not work for 

```kt
@Test
fun canTransformResource() {
  assertThat(transformer.canTransformResource("")).isFalse()
  assertThat(transformer.canTransformResource(".")).isFalse()
  assertThat(transformer.canTransformResource("tmp.dat")).isFalse()
  assertThat(transformer.canTransformResource("$PLUGIN_CACHE_FILE.tmp")).isFalse()
  assertThat(transformer.canTransformResource("tmp/$PLUGIN_CACHE_FILE")).isFalse()
  assertThat(transformer.canTransformResource(PLUGIN_CACHE_FILE)).isTrue()
}
```

Seems we should use `element.relativePath.pathString` here.